### PR TITLE
Easyloggingpp: Add package

### PR DESCRIPTION
**Pull request by VolkerChristian**
Maintainer: [me@vchrist.at](mailto:me@vchrist.at)
Compile tested: OpenWRT-23.05-rc2, arm_cortex-a7_neon-vfpv4, mips_24kc
Run tested: arm_cortex-a7_neon-vfpv4 (Linksys MR8300), mips_24kc (tplink_archer-a7), OpenWrt 23.05-rc2)

Description:
* Add package easyloggingpp

Signed-off-by: Volker Christian <me@vchrist.at>

### CODE REVIEW

**Comment by BKPepe** [comment]
Hi,

- What is the usage for this package on the router itself? 
- Why this PR goes to the branch instead of master branch?
- What is the purpose for this library?  The description is not exactly useful to know.

File: libs/easyloggingpp/Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=easyloggingpp
+PKG_VERSION:=9.97.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (BKPepe) [inline libs/easyloggingpp/Makefile:7]: This is wrong. :-( The tarball is going to be downloaded only as version in the dl folder., so no name and thats not wanted.

File: libs/easyloggingpp/Makefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+define Build/InstallDev
+	$(INSTALL_DIR) $(1)/usr/include
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/include/easylogging++.h $(1)/usr/include
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/include/easylogging++.cc $(1)/usr/include
+	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/pkgconfig/easyloggingpp.pc $(1)/usr/lib/pkgconfig
+endef
+
+# This command is always the last, it uses the definitions and variables we give above in order to get the job done
```
Comment (BKPepe) [inline libs/easyloggingpp/Makefile:37]: Drop

File: libs/easyloggingpp/Makefile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+include $(INCLUDE_DIR)/cmake.mk
+
+# Package definition; instructs on how and where our package will appear in the overall configuration menu ('make menuconfig')
+define Package/easyloggingpp
+  SECTION:=lib
+  CATEGORY:=Libraries
+  TITLE:=Easylogging C++ Library
+endef
+
+# Package description; a more verbose description on what our package does
```
Comment (BKPepe) [inline libs/easyloggingpp/Makefile:24]: Its not exactly verbose description below, right?

File: libs/easyloggingpp/Makefile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+PKG_SOURCE_URL:=https://github.com/abumq/easyloggingpp/archive/refs/tags
+PKG_HASH:=9110638e21ef02428254af8688bf9e766483db8cc2624144aa3c59006907ce22
+PKG_MAINTAINER:=Volker Christian <me@vchrist.at>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+# Package definition; instructs on how and where our package will appear in the overall configuration menu ('make menuconfig')
```
Comment (BKPepe) [inline libs/easyloggingpp/Makefile:17]: Drop

File: libs/easyloggingpp/Makefile [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+define Package/easyloggingpp
```
Comment (BKPepe) [inline libs/easyloggingpp/Makefile:18]: Add URL and most likely some dependencies.

**Review by BKPepe — CHANGES_REQUESTED** [review]

**Comment by VolkerChristian** [comment]
Sorry for being that sloppy in my initial pull request.

- What is the usage for this package on the router itself?This package is not intended to be installed on the router itself. It is just a development library used by for building applications.         The reason I want to add this package to OpenWRT is that finally I want to add a new MQTT-broker and MQTT-integrator application [MQTTbroker](https://github.com/VolkerChristian/mqttbroker) (documentation needs to be done) to OpenWRT.         MQTTbroker relies on the c++ network framework [SNode.C](https://github.com/VolkerChristian/snode.c).       Both, applications and framework rely itself on easylogingpp.      This, before creating a pull request for SNode.C and MQTTBroker easyloggingpp needs to be in OpenWRT
- 
- Why this PR goes to the branch instead of master branch? Hm, I am not really familiar about how the different branches interact. Please advice me if a pull request to master is enough to integrate this package in 23.05.
- 
- What is the purpose for this library? I think you mean the description in the Makefile. This should be fixed now.

**Comment by VolkerChristian** [comment]
I have closed my initial three pull requests for file, nlohmannjson, and easylogging and started over with a clean and fresh set.
Now only sent to master.
Should i also fill a pull request to 23.5?

**Pull request closed without merging**